### DISCUSSION TITLE
Uno R3 + Wifi ATmega328P + ESP8266 (32Mb di Memoria) USB-TTL CH340G per Arduino Uno Nodemcu Wemos ESP8266

### DISCUSSION THREAD

**Discussion by maury77** [arendst/Tasmota#11585] [Q&A] [answered]
Is it possible to install tasmota on this card?

I need to use 3 Analog pins

Uno R3 + Wifi ATmega328P + ESP8266 (32Mb di Memoria) USB-TTL CH340G per Arduino Uno Nodemcu Wemos **ESP8266** 

https://it.aliexpress.com/item/33006588335.html?spm=a2g0o.cart.0.0.92773c00XUXZxP&mp=1

**Comment by sfromis** [comment, marked as answer]
It should be possible to install Tasmota on the ESP8266 part of that board. Using TasmotaClient, you should be able to get it to cooperate with the the ATmega part too if you create a suitable program for that chip. 

If my aim was to get 3 analog pins, I'd find it a lot easier to either get a simpler I2C ADC breakout board (like ADS1115) for the ESP8266 (on something like a NodeMCU dev board), or use the ESP32 (supported by Tasmota) with several analog input pins. Details depend on your expectations about speed, voltages and quality of ADC sampling, of course.

**Reply by ascillato** [reply]
It is possible to use Tasmota on the ESP8266.
The 32Mb is part of a tricky advertisement. It is not 32MegaBytes. It is really 32 MegaBits that are equal to 4 MegaBytes.
The ESP8266 support a max of 16MegaBytes.
A 4 MegaBytes is the same as a normal NodeMCU.
So this board is compatible with Tasmota, but it is more expensive than a NodeMCU or a ESP32 dev board.

**Reply by maury77** [reply]
Ok thanks, what alternatives do you propose to have a nodeMCu with 3 analogic inputs?

**Reply by sfromis** [reply]
I already suggested the ADS1115 breakout board. Do review if it fits your use case. Probably OK if ATmega328P would be OK. Personally, I'd first check if ESP32 analog inputs would be good for the task.
